### PR TITLE
INTL-65: Read and preserve existing messages in intl_message_migration

### PR DESCRIPTION
**Pull request by alanknight-wk**
## Motivation
  
If we run the codemod when an _intl.dart class already exists, read the existing messages and preserve them. This should include any manual modifications, e.g. adding description or meaning.

## Changes
  
Read the file, and partially parse the messages, so we have them organized by name. That also improves the sorting.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/executables/intl_message_migration_test.dart [hunk 1]
```diff
@@ -173,15 +191,21 @@ someMoreStrings() => (mui.Button()
 }
 
 List<String> extraMessages = [
-  "  static String get orange => Intl.message('orange', name: 'TestProjectIntl_orange',);",
-  "  static String get aquamarine => Intl.message('aquamarine', name: 'TestProjectIntl_aquamarine',);"
+  "  static String get orange => Intl.message('orange', name: 'TestProjectIntl_orange');",
```
Comment (alanknight-wk) [inline test/executables/intl_message_migration_test.dart:194]: Change lots of tests not to have the trailing comma.

File: lib/src/intl_suggestors/utils.dart [hunk 2]
```diff
@@ -108,7 +108,7 @@ String intlFunctionBody(
 }) {
   var escapedStr = escapeApos(message);
   String delimiter = (isMultiline) ? "'''" : "'";
-  return "Intl.message(${delimiter}${escapedStr}${delimiter}, ${args.isNotEmpty ? 'args: ${args}, ' : ''}name: '$name',)";
+  return "Intl.message(${delimiter}${escapedStr}${delimiter}, ${args.isNotEmpty ? 'args: ${args}, ' : ''}name: '$name')";
```
Comment (alanknight-wk) [inline lib/src/intl_suggestors/utils.dart:111]: The trailing comma disappears if we regenerate from the AST, which makes lots of test differences, so just don't put it in.

**Comment by alanknight-wk** [comment]
Made changes to support handling duplicate names, which are unfortunately large.

Removed the previous mechanism for numbering conflicting names. Instead the IntlMessages keeps track of the messages we have so far. Syntax generation is moved out of utils into a class associated with that, so when you ask for a definition or call it checks for an existing name that looks different and can increment a suffix number for it.
Also some renaming and tidying.

**Review by robbecker-wf — APPROVED** [review]
+10 tested on a handful of repos

**Comment by robbecker-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**